### ISSUE TITLE
Bug in parallel method

### ISSUE THREAD

**Issue by riophae** [cho45/jsdeferred#26]
```
Deferred.parallel([ 1, 2, 3 ].map(function(i) { 
    var d = new Deferred; 
    setTimeout(function() { 
        d.fail(i); 
    });
    return d;
})).error(console.log.bind(console));
// => 1
// => 2
// => 3
```

Since you will get only one result for a succeeded paralleled-deferred I guess there shouldn't be three results for a failed paralleled-deferred.
